ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -731,6 +734,24 @@ private static void start(SessionState startSs, boolean isAsync, LogHelper conso
       // that would cause ClassNoFoundException otherwise
       throw new RuntimeException(e);
     }
+    try {
+      if(!startSs.isHiveServerQuery){
```

Comment:
A lot of nested if conditionals. Can we move this inside a method and probably invert the if conditions so that if those are satisfied then return. 
Something like - 
```
void methodName() {
       if (startSs.isHiveServerQuery) {
           return;
       }
       if (!getSessionConf().getBoolVar(ConfVars.METASTORE_USE_THRIFT_SASL) {
           return;
       }
       ...... (so on)
}
```

Reply:
> A lot of nested if conditionals. Can we move this inside a method and probably invert the if conditions so that if those are satisfied then return. Something like -
> 
> ```
> void methodName() {
>        if (startSs.isHiveServerQuery) {
>            return;
>        }
>        if (!getSessionConf().getBoolVar(ConfVars.METASTORE_USE_THRIFT_SASL) {
>            return;
>        }
>        ...... (so on)
> }
> ```

Looks good, I will do it this way.